app/components/tags_input/tags_input.js

```diff
@@ -1,18 +1,92 @@
 import { Controller } from 'stimulus';
 import Tagify from '@yaireo/tagify';
 
+const COLORS = ['#F4C317', '#0898FF', '#67D881', '#F4177A'];
+
+const tagColor = tagData => {
+  if (!tagData.id) {
+    return 'var(--color-text)';
+  }
+
+  const COLORS = ['#F4C317', '#0898FF', '#67D881', '#F4177A'];
+  return COLORS[tagData.id % COLORS.length];
+};
+
+function dropdownItemTemplate(tagData) {
+  const membersLabel =
+    tagData.count === 1
+      ? this.settings.labels.members.one
+      : this.settings.labels.members.other;
+
+  const color = tagColor(tagData);
+
+  return `
+    <div ${this.getAttributes(tagData)}
+      class="${this.settings.classNames.dropdownItem} TagsInput-dropdownItem"
+      tabindex="0"
+      role="option"
+    >
+      <span class="TagsInput-name" style="--tag-bg: ${color}">
+        ${tagData.name}
+      </span>
+      <span class="TagsInput-count">${tagData.count} ${membersLabel}<span>
+    </div>
+  `;
+}
+
+function transformTag(tagData) {
+  const color = tagColor(tagData);
+
+  tagData.style = `
+    --tag-bg: ${color};
+    --tag-hover: ${color};
+    --tag-remove-bg: ${color};
+    --tag-remove-btn-bg--hover: ${color};
+  `;
+}
+
 export default class extends Controller {
   static targets = ['input'];
 
   connect() {
-    new Tagify(this.inputTarget, {
-      originalInputValueFormat: tags => tags.map(tag => tag.value).join(','),
-      whitelist: JSON.parse(this.data.get('options')),
+    this.tagify = new Tagify(this.inputTarget, {
+      originalInputValueFormat: tags => {
+        return tags.map(tag => tag.value).join(',');
+      },
+
+      whitelist: JSON.parse(this.data.get('available-tags')),
+      enforceWhitelist: !JSON.parse(this.data.get('allow-new')),
```

Comment:
I have found this [issue in tagify](https://github.com/yairEO/tagify/issues/465) reporting the weird behavior for `request_form` adding them invalidating and removing the tag... Haven't had time to go through it...

Reply:
Thanks, I added that as a separate issue: https://github.com/tactilenews/100eyes/issues/398